### ISSUE TITLE
[Question] Using SB3 with PyTorch C++ frontend (libtorch)

### ISSUE THREAD

**Issue by Xvdgeest** [DLR-RM/stable-baselines3#218]
Is it possible to import SB3 in C++ as an extension to libtorch, or is it planned to add this functionality?

**Comment by araffin** [comment]
Hello,

>Is it possible to import SB3 in C++ as an extension to libtorch

What do you mean exactly?
Is it possible to export python libraries to c++?

However, we have a section in the documentation that needs to be filled: https://stable-baselines3.readthedocs.io/en/master/guide/export.html#export-to-c
It is about exporting trained models to c++ (and it is related to the pytorch jit, see #57 ), note that this is not about training RL models in c++.

**Comment by Xvdgeest** [comment]
Since Pytorch has a C++ frontend: (libtorch)
#include <torch/torch.h>

My question is, is it planned to port SB3 to this frontend, and have the library directly available in cpp, to get syntax like this:

#include <sb3/sb3.h>
sb3::PPO model = sb3::PPO('MlpPolicy', env, verbose=1);

**Comment by araffin** [comment]
>My question is, is it planned to port SB3 to this frontend, and have the library directly available in cpp, to get syntax like this:

That would require to rewrite the complete lib using C++, so not planned for now.
But we would obviously happy if someone does it ;)

**Comment by Xvdgeest** [comment]
Yes that's what I thought. I might port some parts as I need them then. Thanks for the clarification!
